src/pydata_sphinx_theme/assets/styles/extensions/_sphinx_design.scss

```diff
@@ -66,25 +66,40 @@ $all-colors: map.merge($pst-semantic-colors, $extra-semantic-colors);
 
 @mixin create-sd-colors($value, $name) {
   // define the pst variables, so that downstream user overrides will work
+  @debug "Creating color variables for semantic color: #{$name}";
+
   --pst-color-#{$name}: #{$value};
 
   // we are now using a11y-combination to calculate the text color - this is based
   // on the WCAG color contrast guidelines
   --pst-color-#{$name}-text: #{a11y-combination($value)};
 
-  // TODO: highlight seems to be used for buttons @trallard to fix on a11y follow-up work
-  --pst-color-#{$name}-highlight: #{color.adjust($value, $lightness: -15%)};
+  // highlight is used for hover effects on buttons, here we make some fluid scaling
+  // to avoid jarring effects -- we create a local variable that we can access
+  // later to calculate the text color
+  $highlight-color: color.scale($value, $lightness: -15%, $saturation: 5%);
+
+  --pst-color-#{$name}-highlight: #{color.scale(
+      $value,
+      $lightness: -15%,
+      $saturation: 5%
+    )};
 
   // override the sphinx-design variables
   --sd-color-#{$name}: var(--pst-color-#{$name});
   --sd-color-#{$name}-text: var(--pst-color-#{$name}-text);
 
-  // TODO: highlight seems to be used for buttons @trallard to fix on a11y follow-up work
+  // highlight is used for hover effects on buttons, here we make some fluid scaling
+  // to avoid jarring effects
   --sd-color-#{$name}-highlight: var(--pst-color-#{$name}-highlight);
+
+  // calculate the text color for the highlight color
+  --pst-color-#{$name}-highlight-text: #{a11y-combination($highlight-color)};
```

Comment:
Since we already calculate a WCAG conformant text colour for our semantic colours it made sense to do the same for highlight colours that are used for hover